### PR TITLE
Config rework

### PR DESCRIPTION
**Pull request by prarit**
The lab config heirarchy is:
    1. ENV variables (LAB_CORE_TOKEN, LAB_CORE_HOST)
            - if specified, core.token and core.host values in
              config files are not updated.
    2. "dot" . user specified config
            - if specified, lower order config files will not override
              the user specified config
    3.  .config/lab/lab.toml (global config)
    4.  .git/lab/lab/toml (worktree config)

Values from the worktree config will override any global config settings.

### CODE REVIEW

File: internal/config/config.go [hunk 1]
```diff
@@ -202,9 +222,21 @@ func GetToken() string {
...
+//	2. "dot" . user specified config
+//		- if specified, lower order config files will not override
+//		  the user specified config
+//	3.  .config/lab/lab.toml (global config)
+//	4.  .git/lab/lab/toml (worktree config)
+//
+// Values from the worktree config will override any global config settings.
+
+// LoadMainConfig() loads the main config file and returns a tuple of
 //  host, user, token, ca_file, skipVerify
```
Comment (zaquestion) [inline internal/config/config.go:238]:
```suggestion
// LoadMainConfig loads the main config file and returns a tuple of
//  host, user, token, ca_file, skipVerify
//
// The lab config heirarchy is:
//	1. ENV variables (LAB_CORE_TOKEN, LAB_CORE_HOST)
//		- if specified, core.token and core.host values in
//		  config files are not updated.
//	2. "dot" . user specified config
//		- if specified, lower order config files will not override
//		  the user specified config
//	3.  .config/lab/lab.toml (global config)
//	4.  .git/lab/lab/toml (worktree config)
//
// Values from the worktree config will override any global config settings.
```

This was technically broken before too, but let's get a proper godoc comment in place. Convention is to start the comment block with the function name (no `()`)
Ref: https://blog.golang.org/godoc

Comment (prarit) [reply internal/config/config.go:238]: Haha :)  TBH it was meant as a reminder to myself while I debug :)  But yes, this needs to be documented and I have not yet made any README.md changes to the current WIP version.

FWIW: what's getting me is the odd panic from configtest.  I'm scratching my head a bit on this one so if you have any ideas let me know.

Comment (zaquestion) [reply internal/config/config.go:238]: Yep the rest of the stack shows what's happening
```
github.com/spf13/viper.(*Viper).GetString(0x0, 0x101c313, 0x9, 0xc0003a2000, 0x10000c00005f4d8)
	/home/travis/gopath/pkg/mod/github.com/spf13/viper@v1.4.0/viper.go:734 +0x3f
github.com/zaquestion/lab/internal/config.New(0xc0005c71a0, 0x28, 0x11d08c0, 0xc00035a2a0, 0x28, 0xe)
	/home/travis/gopath/src/github.com/zaquestion/lab/internal/config/config.go:44 +0x99
github.com/zaquestion/lab/internal/config.TestNewConfig.func1(0xc00057e780)
	/home/travis/gopath/src/github.com/zaquestion/lab/internal/config/config_test.go:40 +0x265
testing.tRunner(0xc00057e780, 0xc0002c64c0)
	/home/travis/.gimme/versions/go1.15.2.linux.amd64/src/testing/testing.go:1127 +0xef
created by testing.(*T).Run
	/home/travis/.gimme/versions/go1.15.2.linux.amd64/src/testing/testing.go:1178 +0x386
```
Most notably in "github.com/spf13/viper.(*Viper).GetString(**0x0**, "

Means that `config.New` is being called without `MainConfig` being set. Probably a scenario specific to the tests I'm guessing, but perhaps there should be a guard at the top of `New` with something like
```
if MainConfig == nil {
    LoadMainConfig()
}
```

**Comment by prarit** [comment]
@zaquestion , finally!  :)  Take a look and let me know what you think.

I'd like to write some code that allows a user to set *any* command variable in the config.  I'm going to probably do that in another MR though.

**Comment by prarit** [comment]
@bmeneguele this changeset could use some independent testing.  TIA.

**Comment by zaquestion** [comment]
> I'd like to write some code that allows a user to set any command variable in the config. I'm going to probably do that in another MR though.

Yeah that would be an appreciated feature I'm sure, I've wanted to ability to set certain flags to default through config for a while now.

**Comment by bmeneguele** [comment]
> @bmeneguele this changeset could use some independent testing. TIA.

Unfortunately, I wasn't able to touch the changes in the past days heheh, but will do that early next week.

**Comment by prarit** [comment]
> > I'd like to write some code that allows a user to set any command variable in the config. I'm going to probably do that in another MR though.
> 
> Yeah that would be an appreciated feature I'm sure, I've wanted to ability to set certain flags to default through config for a while now.

https://github.com/zaquestion/lab/pull/436 :)

**Review by bmeneguele — APPROVED** [review]
The change I suggested in README is totally a personal opinion. The MR should not be blocked by that :-).
From my point of view and local tests the results looks pretty good. Nice work @prarit .

File: README.md [hunk 2]
```diff
@@ -87,24 +87,30 @@ See the [contribution guide](CONTRIBUTING.md).
...
+information and save it into `~/.config/lab/lab.toml`.
 For example:
 ```
 $ lab
 Enter default GitLab host (default: https://gitlab.com):
 Enter default GitLab token:
 ```
 
-Additionally, there are command-specific configuration files in TOML: .git/lab/[command]-metadata.toml which provide users the ability to override some command line options:
+Command-specific options can be set in the config files.
```
[the remark this answers is not in the document]
Comment (bmeneguele) [reply README.md:109]: OTOH, this suggestion should've been posted to MR #436 and not here.. :facepalm: hahah.

**Pull request merged**